lsp/src/main/java/org/aya/lsp/options/RenderOptions.java

```diff
@@ -0,0 +1,216 @@
+import org.aya.pretty.style.AyaStyleFamily;
+import org.javacs.lsp.MarkupKind;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * @param colorScheme
+ * @param styleFamily
+ */
+public record RenderOptions(
+  @NotNull Option<ColorScheme> colorScheme,
+  @NotNull Option<StyleFamily> styleFamily
+) {
+  public final static RenderOptions DEFAULT = new RenderOptions(
+    Option.some(AyaColorScheme.EMPTY),
+    Option.some(AyaStyleFamily.DEFAULT)
+  );
+
+  public enum RenderTarget {
+    Debug,
+    HTML,
+    TeX
+  }
+
+  public static final @NotNull ImmutableMap<String, ColorScheme> BUILTIN_COLOR_SCHEMES = ImmutableMap.of(
+    "emacs", AyaColorScheme.EMACS,
+    "intellij", AyaColorScheme.INTELLIJ,
+    "none", AyaColorScheme.EMPTY
+  );
+
+  /// region Helper
+
+  /**
+   * Construct a {@link RenderOptions} from {@link ServerOptions}
+   */
+  @Contract(pure = true)
+  public static @NotNull Result<@NotNull RenderOptions, @NotNull String> fromServerOptions(@NotNull ServerOptions options) {
+    Option<ColorScheme> colorScheme = Option.none();
+
+    // colorScheme is none if both colorName and colorOverride are null
+    if (options.colorName != null || options.colorOverride != null) {
+      Result<ColorScheme, String> parsed = parseColorScheme(options);
+      if (parsed.isErr()) return Result.err(parsed.getErr());
+
+      colorScheme = Option.some(parsed.get());
+    }
+    // TODO[hoshino]: styleFaimily
+
+
+    return Result.ok(new RenderOptions(colorScheme, Option.none()));
+  }
+
+  /**
+   * @apiNote this function returns {@code Result.ok(AyaColorScheme.EMPTY)} even though {@code options.colorName and options.colorOverride are null}
+   */
+  private static Result<ColorScheme, String> parseColorScheme(@NotNull ServerOptions options) {
+    var colorName = options.colorName;
+    var baseOption = options.colorName == null
+      ? Option.some(AyaColorScheme.EMPTY)
+      : BUILTIN_COLOR_SCHEMES.getOption(colorName);
+
+    if (baseOption.isEmpty()) {
+      return Result.err("Invalid color scheme name");
+    }
+
+    var base = baseOption.get();
+    // We won't modify the origin ColorScheme
+    var newColorScheme = MutableMap.from(base.definedColors());
+    var override = options.colorOverride;
+    if (override == null || override.isEmpty()) return Result.ok(base);
+
+    boolean modified = false;
+
+    for (var entry : override.entrySet()) {
+      var key = entry.getKey();
+      var value = entry.getValue();
+
+      var isKeyValid = key != null && Seq.from(AyaColorScheme.Key.values()).view()
+        .map(AyaColorScheme.Key::key)
+        .contains(key);
+      var isValueValid = value != null;   // I don't like to inline this code, sorry!
+
+      if (!isKeyValid) return Result.err("Invalid key: " + key);
+      if (!isValueValid) return Result.err("Invalid color: null");
+
+      var colorResult = parseColor(value);
```

Comment:
Just ignore them instead of aborting the whole parser for future compatibility. The code could also be simplfiied in this case.